x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.test.ts

```diff
@@ -1202,4 +1203,25 @@ describe('utils', () => {
       expect(output).toEqual(expected);
     });
   });
+
+  describe('getDuplicates', () => {
+    test("returns a string showing the duplicate keys of 'value2' and 'value3'", () => {
+      const output = getDuplicates({
+        value1: 1,
+        value2: 2,
+        value3: 2,
+      });
+      const expected = 'value2, value3';
+      expect(output).toEqual(expected);
+    });
+    test('returns null when given a map of no duplicates', () => {
+      const output = getDuplicates({
+        value1: 1,
+        value2: 1,
+        value3: 1,
+      });
+      const expected = null;
+      expect(output).toEqual(expected);
+    });
+  });
```

Comment:
Awesome! 👍 Love the tests! Going to pay off so much in the long run. Hopefully they give you the warm fuzzy feelings that what you're writing is some great quality stuff.

Your code is really starting to shine 🌟 with how you write it. Future maintainers, maybe even future Devin, will be nodding their head in approval if these ever break, like thanks past Devin you got my back!